pkg/frontend/mysql_cmd_executor.go

```diff
@@ -1694,13 +1719,28 @@ func (cwft *TxnComputationWrapper) GetColumns() ([]interface{}, error) {
 	cols := plan2.GetResultColumnsFromPlan(cwft.plan)
 	switch cwft.GetAst().(type) {
 	case *tree.ShowColumns:
-		cols = []*plan2.ColDef{
-			{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Field"},
-			{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Type"},
-			{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Null"},
-			{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Key"},
-			{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Default"},
-			{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Comment"},
+		if len(cols) == 7 {
+			cols = []*plan2.ColDef{
+				{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Field"},
+				{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Type"},
+				{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Null"},
+				{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Key"},
+				{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Default"},
+				{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Extra"},
+				{Typ: &plan2.Type{Id: int32(types.T_char)}, Name: "Comment"},
+			}
+		} else {
+			cols = []*plan2.ColDef{
```

Comment:
If we always let len(cols) == 8, and when build sql we simply project different number of columns, will that work?

Reply:
there should be 9 columns when full is specified, and 7 columns otherwise.
but we can not judge whether there is a full keyword here, it's not passed here.
i think this is the simpliest way.